modules/articles/server/controllers/articles.server.controller.js

```diff
@@ -81,14 +81,45 @@ exports.delete = function (req, res) {
  * List of Articles
  */
 exports.list = function (req, res) {
-  Article.find().sort('-created').populate('user', 'displayName').exec(function (err, articles) {
-    if (err) {
-      return res.status(400).send({
-        message: errorHandler.getErrorMessage(err)
-      });
-    } else {
-      res.json(articles);
-    }
+  // Prepare paging parameters
+  var page = Math.max(0, req.query.page - 1); // using a zero-based page index for use with skip()
+  var take = req.query.take || 10;
+
+  /** 
+  * I don't like having the filter, so I'm removing it for now. It's going to introduce a bit of complexity. 
+  * For instance, when we're querying the collection statistics, to get the total count used with the paging.
+  * We may have discrepencies between the total amount of documents in the collection, and what this filter will return.
+  * An alternative option would be to return the count of documents returned from this query/filter. However, this would
+  * require us to query first for the count & then again using our limit() & skip() conditions.
+  *
+  * Something in the form of:
+  *  { articles: [], count: x }
+  *
+  * The idea behind the filter is that mongoose will start at the beginning of the collection, and move it's way up to 
+  * satisfy the skip condition. This could have performance issues. ref: http://stackoverflow.com/questions/5539955/how-to-paginate-with-mongoose-in-node-js/23640287#23640287
+  */
+
+  var today = new Date(Date.now());
+  // only articles from the past 30 days
+  var dateFloor = (new Date()).setDate(today.getDate() - 30);
+  var filter = { created: { $gte: dateFloor } };
+
+  // cap the amount of articles queried
+  var query = Article.find(filter).limit(500).sort('-created');
```

Comment:
500 might too large for a default amount of Articles to be querying. I'm unsure of where we might want to set this at. Obviously, this default isn't relevant until the amount of documents reached this limit. Our user's requirement will vary quite a bit.

With that said, perhaps 100 is better?